src/SeleniumLibrary/keywords/screenshot.py

```diff
@@ -112,6 +112,15 @@ def capture_page_screenshot(self,
         | File Should Exist | ${OTHER_DIR}${/}sc-000001.png |
 
         """
+        try:
+            # try to access the browser property
+            b = self.browser
+            
+        except RuntimeError:
+            # we got an error, just exit
+            self.info("Couldn't capture page screenshot because no browser is opened")
+            return
+        
```

Comment:
Comments and empty lines are unnecessary and somewhat against the style used elsewhere in this codebase. I'd also say it's clearer to just have

```python
try:
    self.browser
except RuntimeErrror:
  # ...
```

than

```python
try:
    b = self.browser
except RuntimeErrror:
  # ...
```
when the `b` variable isn't used for anything. Could also consider using `try/except/else` instead of returning from `except`.

It might actually also be a good idea to create new `BrowserNotOpen` exception and the `browser` property to use it instead of the generic `RuntimeError`. The code here would then look like

```python
try:
    self.browser
except BrowserNotOpen:
  # ...
```

which would be really clean. Not sure is it worth it, though.

Reply:
Good suggestions. I'll get back to this on Monday